src/zentables/zentables.py

```diff
@@ -840,53 +832,63 @@ def _combine_n_pct(
 
 
-def _is_between(x, ceiling: int = 10):
+def _is_between(val, low: int = 1, high: int = 10):
     """
     Helper function that verifies if the input is below ceiling value
     """
-    return x in range(1, ceiling)
+    return low <= val < high
 
 
-def _add_random_noise(x, rng):
-    """
-    Helper function that adds random noise to entry according a Uniform distribution
-    """
-    return x + rng.uniform(0, 1)
+def _seed_to_rng(seed: Optional[Union[int, Generator]] = None) -> Generator:
+    if seed is None:
+        return np.random.default_rng()
+    if isinstance(seed, int):
+        return np.random.default_rng(seed)
 
 
-def _local_suppression(mini_df_n: pd.DataFrame, ceiling: int = 10, seed: int = 2021):
+def _local_suppression(
+    mini_df_n: pd.DataFrame,
+    low: int = 1,
+    high: int = 10,
+    seed: Optional[Union[int, Generator]] = None,
+):
     """
-    Helper function that applies cell suppression to
+    Helper function that applies cell suppression to mini_df_n.
+    mini_df_n is assumed to be a smaller portion of a larger
     """
-    rng = np.random.default_rng(seed)
-    mask = mini_df_n.applymap(lambda entry: _is_between(entry, ceiling))
+    rng = _seed_to_rng(seed)
+    mask = np.logical_and(mini_df_n >= low, mini_df_n < high)
+    tie_beaking_number = mini_df_n.abs().max().max() * 2
 
     while True:
         stop_suppression = False
         roi = np.where(mask.sum(axis=1) == 1)[0]
-        if len(roi) > 1:
-            assert stop_suppression == False
-        else:
-            # There are no longer rows that need to be suppressed.
-            stop_suppression = True
+        stop_suppression = len(roi) == 0
 
-        tie_breaker_df = (
-            mini_df_n.applymap(lambda entry: _add_random_noise(entry, rng))
-            + mask * 10e10
-        )
-        min_cols = tie_breaker_df.idxmin(axis=1)
-        # Suppress the identified minimums per row.
-        for pair in list(zip(mask.index[roi], min_cols.iloc[roi])):
-            mask.loc[pair] = True
+        if not stop_suppression:
+            tie_breaker_df = (
+                mini_df_n
+                + rng.uniform(0, 1e-2, size=mini_df_n.shape)
+                + (mask * tie_beaking_number).astype("float")
+            )
+            min_cols = tie_breaker_df.idxmin(axis=1)
+            # Suppress the identified minimums per row.
+            # for pair in list(zip(mask.index[roi], min_cols.iloc[roi])):
+            #     mask.loc[pair] = True
+            mask.loc[mask.index[roi], min_cols.iloc[roi]] = True
 
         coi = np.where(mask.sum(axis=0) == 1)[0]
-        tie_breaker_df = (
-            mini_df_n.applymap(lambda entry: _add_random_noise(entry, rng))
-            + mask * 10e10
-        )
-        min_rows = tie_breaker_df.idxmin(axis=0)
-        for pair in list(zip(min_rows.iloc[coi], mask.columns[coi])):
-            mask.loc[pair] = True
+        stop_suppression = len(coi) == 0
+        if not stop_suppression:
```

Comment:
So I think I lied in my original logic. I think you to do:

```python
if len(coi) > 0:
```

here and then instead of `if stop_suppression` below you want to say:

```python
if len(coi) == 0 and len(roi) == 0:
    break
```